shell/platform/windows/compositor_opengl.cc

```diff
@@ -0,0 +1,200 @@
+  FML_DCHECK(is_initialized_);
+  FML_DCHECK(store->type == kFlutterBackingStoreTypeOpenGL);
+  FML_DCHECK(store->open_gl.type == kFlutterOpenGLTargetTypeFramebuffer);
+
+  auto user_data = static_cast<FramebufferBackingStore*>(
+      store->open_gl.framebuffer.user_data);
+
+  gl_->DeleteFramebuffers(1, &user_data->framebuffer_id);
+  gl_->DeleteTextures(1, &user_data->texture_id);
+
+  delete user_data;
+  return true;
+}
+
+bool CompositorOpenGL::Present(const FlutterLayer** layers,
+                               size_t layers_count) {
+  if (!engine_->view()) {
+    return false;
+  }
+
+  // Clear the view if there are no layers to present.
+  if (layers_count == 0) {
+    // Normally the compositor is initialized when the first backing store is
+    // created. However, on an empty frame no backing stores are created and
+    // the present needs to initialize the compositor.
+    if (!is_initialized_ && !Initialize()) {
+      return false;
+    }
+
+    return ClearSurface();
+  }
+
+  // TODO: Support compositing layers and platform views.
+  // See: https://github.com/flutter/flutter/issues/31713
+  FML_DCHECK(is_initialized_);
+  FML_DCHECK(layers_count == 1);
+  FML_DCHECK(layers[0]->offset.x == 0 && layers[0]->offset.y == 0);
+  FML_DCHECK(layers[0]->type == kFlutterLayerContentTypeBackingStore);
+  FML_DCHECK(layers[0]->backing_store->type == kFlutterBackingStoreTypeOpenGL);
+  FML_DCHECK(layers[0]->backing_store->open_gl.type ==
+             kFlutterOpenGLTargetTypeFramebuffer);
+
+  auto width = layers[0]->size.width;
+  auto height = layers[0]->size.height;
+
+  // Check if this frame can be presented. This resizes the surface if a resize
+  // is pending and |width| and |height| match the target size.
+  if (!engine_->view()->OnFrameGenerated(width, height)) {
+    return false;
+  }
+
+  if (!engine_->surface_manager()->MakeCurrent()) {
+    return false;
+  }
+
+  auto source_id = layers[0]->backing_store->open_gl.framebuffer.name;
+
+  gl_->Disable(GL_SCISSOR_TEST);
+  gl_->Disable(GL_DEPTH_TEST);
+  gl_->Disable(GL_STENCIL_TEST);
+
+  gl_->BindFramebuffer(GL_READ_FRAMEBUFFER, source_id);
+  gl_->BindFramebuffer(GL_DRAW_FRAMEBUFFER, kWindowFrameBufferId);
+
+  gl_->BlitFramebuffer(0,                    // srcX0
+                       0,                    // srcY0
+                       width,                // srcX1
+                       height,               // srcY1
+                       0,                    // dstX0
+                       0,                    // dstY0
+                       width,                // dstX1
+                       height,               // dstY1
+                       GL_COLOR_BUFFER_BIT,  // mask
+                       GL_NEAREST            // filter
+  );
+
+  gl_->BindFramebuffer(GL_READ_FRAMEBUFFER, 0);
+  gl_->BindFramebuffer(GL_DRAW_FRAMEBUFFER, 0);
+
+  gl_->Enable(GL_SCISSOR_TEST);
+  gl_->Enable(GL_DEPTH_TEST);
+  gl_->Enable(GL_STENCIL_TEST);
```

Comment:
This is new:

1. Re-enable scissor, depth, and stencil tests
1. Previously we were unbinding the read/draw framebuffers. This doesn't appear to be strictly necessary, but from my understanding being explicit may prevent future bugs

Reply:
Usually GL renderers will set all necessary state before performing a set of relevant operations. Doubly so if you're disabling a feature. So I'd skip this attempt at restoring?

Reply:
From an offline conversation with Brandon:

> GL_SCISSOR_TEST is off by default, and if it's needed for some other framebuffer operation in the future, code that intends to use the scissor test should be explicitly turning it on just before it's needed anyhow.

Reply:
Marking this as resolved as this restore logic was removed.